x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/PushFiltersToSource.java

```diff
@@ -252,9 +251,7 @@ static boolean canPushToSource(Expression exp, LucenePushdownPredicates lucenePu
                 && Expressions.foldable(cidrMatch.matches());
         } else if (exp instanceof SpatialRelatesFunction spatial) {
             return canPushSpatialFunctionToSource(spatial, lucenePushdownPredicates);
-        } else if (exp instanceof Term term) {
-            return term.field() instanceof FieldAttribute && DataType.isString(term.field().dataType());
-        } else if (exp instanceof FullTextFunction) {
+        } else if (exp instanceof TranslationAware) {
```

Comment:
1. remove the check for `Term` - the same way we removed it for `Match`. This check is not needed since `Term` (the same as `Match`) implements the `Validatable` interface.
2. use `TranslationAware` instead of `FullTextFunction`. In the initial proposal `TranslationAware` has a `boolean translatable()` method. We don't have that yet - this is also a check at the data node level and we expect that if a `TranslationAware` expression cannot be pushed down, that would have been caught much earlier at the verifier level.

Reply:
If we're not using yet `TranslationAware` directly in the codebase, why make this change? 🤔 

For me it's good to keep using `FullTextFunction` until `TranslationAware` is effectively used.

Reply:
`TranslationAware` is effectively used:

https://github.com/elastic/elasticsearch/blob/24773e0ba619e8f99293521b389a7266a0c8214d/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/FullTextFunction.java#L36

what we don't have is that `boolean translatable()` method and I think for now we don't need it, but we might in the future 🤷‍♀️

Reply:
> what we don't have is that boolean translatable() method and I think for now we don't need it, but we might in the future 🤷‍♀️

sorry, that's what I meant - we just have the interface as a marker and the interface method is not actually used. I don't see the advantage on making the switch until the interface method is actually used.

Reply:
okay - I can revert this for now - no issue - and we'll come back to it later